src/sentry/migrations/0315_add_type_to_group.py

```diff
@@ -0,0 +1,55 @@
+# Generated by Django 2.2.28 on 2022-08-15 23:06
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = True
+
+    # This flag is used to decide whether to run this migration in a transaction or not. Generally
+    # we don't want to run in a transaction here, since for long running operations like data
+    # back-fills this results in us locking an increasing number of rows until we finally commit.
+    atomic = False
+
+    dependencies = [
+        ("sentry", "0314_bit_int_for_org_and_project_id"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrations.RunSQL(
+                    """
```

Comment:
@wedamija I'm not sure if I've created the custom SQL correctly, is it meant to be the same in both? I did run this locally and it works, I kind of hodgepodged together the instructions [here](https://develop.sentry.dev/database-migrations/#adding-columns-with-a-default) with your comments in the related Jira ticket.

Reply:
The `database_operations` look good. 

For `state_operations` we should use the `migrations.AddField` method instead of `RunSQL`. We want to do this because the purpose of `state_operations` is to build up the state of the database in Django migrations so that we can compare it to existing models. It's how Django knows when to generate new migrations when you run the `makemigrations` command.

You can either build this manually, or even easier is to generate the migration you would need to create this column normally, and then paste those generated operations into `state_operations`. This makes them no-ops from the db's point of view, but makes sure that the state matches the changes to the model.